rest/route/sns.go

```diff
@@ -391,17 +404,63 @@ func (sns *ecsSNS) handleNotification(ctx context.Context, notification ecsEvent
 			// stopped.
 			return nil
 		case cocoa.StatusStopped:
-			reason := notification.Detail.StoppedReason
+			reason := detail.StoppedReason
 			if reason == "" {
 				reason = "stopped due to unknown reason"
 			}
 			return sns.handleStoppedPod(ctx, p, fmt.Sprintf("SNS notification: %s", reason))
 		default:
 			return gimlet.ErrorResponse{
 				StatusCode: http.StatusBadRequest,
-				Message:    fmt.Sprintf("unrecognized status '%s'", notification.Detail.LastStatus),
+				Message:    fmt.Sprintf("unrecognized status '%s'", detail.LastStatus),
 			}
 		}
+	case ecsContainerInstanceStateChangeType:
+		detail, ok := notification.Detail.(ecsContainerInstanceEventDetail)
+		if !ok {
+			return errors.New("notification details should be ECS task event details")
+		}
+		if ecs.ContainerInstanceStatus(detail.Status) != ecs.ContainerInstanceStatusDraining {
+			return nil
+		}
+
+		grip.Info(message.Fields{
+			"message":            "got notification for draining container instance",
+			"container_instance": detail.ContainerInstanceARN,
+			"cluster":            detail.ClusterARN,
+			"reason":             detail.StatusReason,
+		})
+
+		taskARNs, err := sns.listECSTasks(ctx, detail)
+		if err != nil {
+			return errors.Wrapf(err, "listing ECS tasks associated with container instance '%s'", detail.ContainerInstanceARN)
+		}
+		for _, taskARN := range taskARNs {
+			p, err := pod.FindOneByExternalID(taskARN)
+			if err != nil {
+				return errors.Wrapf(err, "finding pod with external ID '%s'", taskARN)
+			}
+			if p == nil {
+				grip.Error(message.Fields{
+					"message": "Found an ECS task running in a draining container instance, but it's not associated with any known pod." +
```

Comment:
Is it worth having an alert for this case?

Reply:
I think it's worth being aware if/when it happens, because it's a case that breaks the assumption that Evergreen is properly tracking is cloud resources. Whenever we have a state mismatch between Evergreen and the cloud provider, we have to keep an eye on such things because they generally represent bugs, races, or leaked resources.

My best guess is that this can hypothetically be triggered in a race condition when first creating the pod. I don't have a good idea of how often it will actually occur in practice, but if it's common, we'll have to add more complex logic to deal with the race.